### DISCUSSION TITLE
Docs resolving urls relatively instead of absolutely

### DISCUSSION THREAD

**Discussion by Ohad31415** [fastapi/fastapi#3676] [Questions]
My question is derived from my recent messing up with root_path for application both behind a proxy + dsn mapping to send requests with some prefix to the proxy. The proxy here is aws api gateway.
I can make the app to work with requests via the dns mapping or straight through the proxy, _however_ I can make the docs ui work only for one of them since each one of them have a different root_path. 

The main problem is with the docs ui needing to fetch the openapi.json so root_path allows it to know how to construct the correct url. 
However wouldn't it be possible to do a relative url resolution something like -
```js
url = location.href
rootPath = url.slice(0, url.lastIndexOf(docsPath))
openapiUrl = `${root}/${openapiPath}
```
And further then all the requests can be made with this `rootPath` prefix. 
So basically the docs template needs the two parameters `docsPath` to know how to find root and `openapiPath`.  
I think that this way would solve my case. 

Does it have any disadvantages or potential problems? 🤔
